File: app/src/main/java/com/star_zero/example/androidmvvm/presentation/tasks/adapter/ItemTaskViewModel.java
package com.star_zero.example.androidmvvm.presentation.tasks.adapter;

import android.databinding.ObservableField;
import android.view.View;
import android.widget.CompoundButton;

import com.star_zero.example.androidmvvm.domain.task.Task;

import org.greenrobot.eventbus.EventBus;

public class ItemTaskViewModel {

    public final ObservableField<Task> task = new ObservableField<>();

    void setTask(Task task) {
        this.task.set(task);
    }

    public void clickItem(View view) {
        EventBus.getDefault().post(new ClickTaskItemEvent(task.get()));
    }

    public void changeCheckComplete(CompoundButton view, boolean isChecked) {
        if (isChecked != task.get().isCompleted()) {
            EventBus.getDefault().post(new ChangeCompleteStateEvent(task.get(), isChecked));
        }
    }

    // ----------------------
    // EventBus event class
    // ----------------------

    public static class ClickTaskItemEvent {
        public final Task task;

        ClickTaskItemEvent(Task task) {
            this.task = task;
        }
    }

    public static class ChangeCompleteStateEvent {
        public final Task task;
        public final boolean completed;

        ChangeCompleteStateEvent(Task task, boolean completed) {
            this.task = task;
            this.completed = completed;
        }
    }
}

File: app/src/main/kotlin/com/star_zero/example/androidmvvm/presentation/tasks/adapter/ItemTaskViewModel.kt
package com.star_zero.example.androidmvvm.presentation.tasks.adapter

import android.databinding.BaseObservable
import android.databinding.Bindable
import android.view.View
import android.widget.CompoundButton

import com.android.databinding.library.baseAdapters.BR
import com.star_zero.example.androidmvvm.domain.task.Task

import org.greenrobot.eventbus.EventBus

class ItemTaskViewModel : BaseObservable() {

    // ----------------------
    // Binding
    // ----------------------

    @get:Bindable
    var task: Task? = null
        set(task) {
            field = task
            notifyPropertyChanged(BR.task)
        }

    fun clickItem(@Suppress("UNUSED_PARAMETER") view: View) {
        val task = this.task ?: return
        EventBus.getDefault().post(ClickTaskItemEvent(task))
    }

    fun changeCheckComplete(@Suppress("UNUSED_PARAMETER") view: CompoundButton, isChecked: Boolean) {
        val task = this.task ?: return
        if (isChecked != task.completed) {
            EventBus.getDefault().post(ChangeCompleteStateEvent(task, isChecked))
        }
    }

    // ----------------------
    // EventBus event class
    // ----------------------

    class ClickTaskItemEvent(val task: Task)

    class ChangeCompleteStateEvent(val task: Task, val completed: Boolean)
}
